source/common/network/ip_address.cc

```diff
@@ -8,16 +8,16 @@
 namespace Envoy {
 namespace Network {
 
-const std::string& IPAddressObject::key() {
-  CONSTRUCT_ON_FIRST_USE(std::string, "envoy.network.ip");
-}
+namespace {
+const std::string& key() { CONSTRUCT_ON_FIRST_USE(std::string, "envoy.network.ip"); }
```

Comment:
constexpr absl::string_view would be fine in .cc file.

Reply:
All factory use `std::string` as string type. Although I think I prefer to return `absl::string_view` then all name() could be used to return the copy of `constexpr absl::string_view`. We can remove most of this `CONSTRUCT_ON_FIRST_USE`.

Reply:
+1 to what @wbpcode said.
There shouldn't be a problem returning absl::string_view, other than cases that build the name dynamically (not sure if there are such cases).
In any case, this should be done in a different PR :)